### PR TITLE
X is a regular element in the ring of list-based polynomials

### PR DESCRIPTION
**Pull request by felixwellen**
This means that multiplication with X is injective.
Depends on #917

### CODE REVIEW

File: Cubical/Algebra/Polynomials/UnivariateList/Properties.agda [hunk 1]
```diff
@@ -639,6 +639,38 @@ module PolyModTheory (R' : CommRing ℓ) where
...
 
+---------------------------------------------------------------------------
+-- The indeterminate is a regular element, i.e. multiplication with it is injective
+
+  prod-X-injective : (p : Poly R') → prod-Xn 1 p ≡ 0P → p ≡ 0P
+  prod-X-injective p Xp≡0 = p≡0
+    where
+      shift = shiftFun (Poly→PolyFun p)
+
+      0r∷p≡shift : Poly→PolyFun (0r ∷ p) ≡ shift
```
Comment (MatthiasHu) [inline Cubical/Algebra/Polynomials/UnivariateList/Properties.agda:650]: If `shiftFun` is exported, then perhaps this lemma (read as a property of `shiftFun`) should also be exported. Do you agree?

**Review by MatthiasHu — COMMENTED** [review]
I like it very much!
One suggestion below.

File: Cubical/Algebra/Polynomials/UnivariateList/Properties.agda (hunk 1 above)
Comment (felixwellen) [reply Cubical/Algebra/Polynomials/UnivariateList/Properties.agda:650]: Yes, I agree...

Comment (felixwellen) [reply Cubical/Algebra/Polynomials/UnivariateList/Properties.agda]: Done

**Review by MatthiasHu — APPROVED** [review]

**Pull request merged**